config.go

```diff
@@ -108,31 +117,37 @@ func (c NginxConfigWriter) Write(layerPath, workingDir, cnbPath string) (string,
 		return "", err
 	}
 
-	f, err := os.OpenFile(filepath.Join(workingDir, "nginx.conf"), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, os.ModePerm)
+	path := filepath.Join(workingDir, "nginx.conf")
+	err = os.WriteFile(path, b.Bytes(), 0600)
```

Comment:
Why does the `nginx.conf` have 0600 permissions, but the `.php.fpm.bp/nginx-fpm.conf` on line 164 have 0660?

Reply:
Good catch. I was relying on some coincidental configuration of `umask` on my machine that I should instead explicitly force with `chmod`.

As for the explanation, we should really only set `0x00` permissions on files using `os.WriteFile`, where x is the permission bit for the current user. Then we should use `os.Chmod` to force the permissions to a specific value. The reason for doing this is that `os.WriteFile` uses `umask` to mask the permissions bits when creating files. This could mean that even if you say `os.WriteFile(path, bytes, 0777)`, you could see the file get permissions of `0755` applied if you have `0022` set as the `umask`.

I think this StackOverflow post helps explain the phenomenon: https://stackoverflow.com/questions/50257981/ioutils-writefile-not-respecting-permissions

Reply:
This is so helpful!! Thank you for the details, I hadn't heard of that before

Reply:
Relatedly, `info.Mode()|0040` this line adds the group read permissions onto the existing permissions?

Reply:
yes. this is a bitwise OR operation. It will set the "read bit" for the group to 1.